src/components/utils/src/threads/thread_posix.cc

```diff
@@ -275,10 +278,13 @@ void Thread::join() {
 }
 
 Thread::~Thread() {
-  finalized_ = true;
-  stopped_ = true;
-  join();
-  pthread_join(handle_, NULL);
+  if (!finalized_) {
+    ThreadCounter::Decrement();
+    finalized_ = true;
+    stopped_ = true;
```

Comment:
@OHerasym, stopped_ = true; will be set again to true in join() -> stop() -> stopeed_ = true. What is the intention to set it true here?